src/Calypso-SystemPlugins-SUnit-Browser/ClyJumpToTestClassCommand.class.st

```diff
@@ -0,0 +1,54 @@
+"
+Description
+--------------------
+
+I am a command available in the class context menu of Calypso. When activated I'll jump on the test class corresponding to the class. 
+
+If the class does not exists I will create it in the corresponding package.
+	
+Internal Representation and Key Implementation Points.
+--------------------
+
+    Instance Variables
+	testClass:		<aClass>		Test class corresponding to the class I was activated with
+"
+Class {
+	#name : #ClyJumpToTestClassCommand,
+	#superclass : #SycSingleClassCommand,
+	#traits : 'TClyGenerateTestClass',
+	#classTraits : 'TClyGenerateTestClass classTrait',
+	#instVars : [
+		'testClass'
+	],
+	#category : #'Calypso-SystemPlugins-SUnit-Browser-Commands'
+}
+
+{ #category : #activation }
+ClyJumpToTestClassCommand class >> fullBrowserMenuActivation [
+	<classAnnotation>
+	
+	^ CmdContextMenuActivation for: ClyClassContextOfFullBrowser
+]
+
+{ #category : #activation }
+ClyJumpToTestClassCommand class >> fullBrowserShortcutActivation [
+	<classAnnotation>
+	
+	^ CmdShortcutActivation by: $g meta, $j meta for: ClyClassContextOfFullBrowser
+]
+
+{ #category : #execution }
+ClyJumpToTestClassCommand >> applyResultInContext: aToolContext [
+	super applyResultInContext: aToolContext.
+	testClass ifNotNil: [ :class | aToolContext showClass: class ]
+]
+
+{ #category : #accessing }
+ClyJumpToTestClassCommand >> defaultMenuItemName [
+	^ 'Jump to test class'
+]
+
+{ #category : #execution }
+ClyJumpToTestClassCommand >> execute [
+	testClass := self testClassFor: targetClass
```

Comment:
Notice that commands were designed to be independent from any context. Idea is to be able create command instance in playground, initialize state using normal methods (accessors/constructors) and execute it. Context methods (prepare, apply) are only a glue code to polymorphically bind commands to tools.  
So here command is called JumpToTest and it should assume that #execute method will perform actual jump to test. But it only generates a test class. According to described logic command should be either renamed to ClyGenerateTestClassCommand or define extra variable #browser which will perform a jump.

In first case #applyResultInContext: would be perfectly logical action: command generates a class and then as a side context specific action it asks context to show this class. (In playground scenario it does not matter because context is not defined).

In second case variable #browser is required because it is the purpose of command to operate the browser. And actual jump will be performed in #execute method. Class generation in that case is a side effect when test is not exist yet. 
Variable #browser in that case should be initialized in #prepare.. method (browser := aContext browser).